*Hello* **Dr. Albert**
# Assignment for week 11 - GIAB

### Overview
The aim of this week assignment is to downlaod the reference genome for a specific region of the genome and evaluate the alignment quality of the reads generated by different sequencing technologies/machines (Option 1).

The Genome In A Botttle (GIAB) is a project to create benchmarked human reference genomes so different labs can validate, compare, and standardize theor sequenicing and variant-calling pipelines against a trusted source. Cancer GIAB has the same goal but with tumor-normal pairs. For whole Genome sequencing, the project used technologies that situated in 4 different approaches: short-read, long-read, single cell, and chromatin configuration capture. In this assignment we want tu compare the technologies for short-reead (Illumina Novaseq, Element Aviti, Pacbio Onso, Ultima standard, Ultima ppmSeq) and long-read sequencing (ONT-standard, ONT-UL E8.2, ONT-UL E8.2.1, PacBop Revio, Pacbio Vega).

So, the approach we took for this evaluation is first to make a design.csv file that contains the web address for downloading the reference from each sequencing technology. So, in the design file you will fin the name of the technologies in the first column and the RUL for the BED files of each of the technologies. Then using parallel for the design file to go over our Make file, which contains the code for downloading the reference genomes and the reads from the project. At the end we can extract the quality information of each read to our reference genome to evaluate which technology produce better reads.

The region of our interest is containing the *CDKN2A* gene which located on chromose 9 short arm 9p21.3 (chr9:21,967,752-21,995,324 for latest assembly in GRCh38/hg38). This gene contains 27,573 bases and encodes the Cyclin Dependent Kinase Inhibitor 2A. this gene produce three transcript variants that differ in their first exons. Two of its splice variants either inhibit CDK4 to block the cell cycle G1 progeression or the other splice variant produce an alternate open reading frame which stabilizes p53 (a famous tumor supressor protein) by restraining MDM2, which is responsible for degradation of p53. The deleted or mutated forms of this gene found in many tumors, especially in pancreatic cancers (Melanoma-Pancreatic Cancer Syndrome) 

One point to consider is that we did not download the reference genome for this week assignment, instead we used the buil-in genome in IGV. So, on the top left of OGV window, you will select **Genomes** > **Download Hosted Genome** and then from the menu select hg38. once it opened, select chromosome 9, where our gene of interest located.

some tips:
a) Some technologies just used in normal tissues while some just in tumor tissue.
b) both of the ultima machines (standard and ppmseq) are not included in the analysis because generate generat cram file instead of bam file.
c) short reads compare the sequences of normal duodenal (N-D) tissues, while long reads compare the tumor (T) samples.
d)pacbio vega for the long read sequences is not included in the analysis, because I could not find which of the pacbio bam files is for the Vega machine, there was not such info in the readme files of the two pacbio reads.